presto-main/src/test/java/com/facebook/presto/server/security/oauth2/TestOAuth2Config.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security.oauth2;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.facebook.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
+import static com.facebook.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
+import static com.facebook.airlift.configuration.testing.ConfigAssertions.recordDefaults;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class TestOAuth2Config
+{
+    @Test
+    public void testDefaults()
+    {
+        assertRecordedDefaults(recordDefaults(OAuth2Config.class)
+                .setStateKey(null)
+                .setIssuer(null)
+                .setClientId(null)
+                .setClientSecret(null)
+                .setScopes("openid")
+                .setChallengeTimeout(new Duration(15, MINUTES))
+                .setPrincipalField("sub")
+                .setGroupsField(null)
+                .setAdditionalAudiences("")
+                .setMaxClockSkew(new Duration(1, MINUTES))
+                .setUserMappingPattern(null)
+                .setUserMappingFile(null)
+                .setEnableRefreshTokens(false)
+                .setEnableDiscovery(true));
+    }
+
+    @Test
+    public void testExplicitPropertyMappings()
+            throws IOException
+    {
+        Path userMappingFile = Files.createTempFile(null, null);
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("http-server.authentication.oauth2.state-key", "key-secret")
```

Comment:
I think we need a dedicated Oauth documentation page, [similar to Trino](https://github.com/trinodb/trino/blob/9e87db0b3caadd3ccc59b5db40026d1a5935c5da/docs/src/main/sphinx/security/oauth2.md) to describe these properties